pkg/controller/constraint/stats_reporter.go

```
@@ -0,0 +1,78 @@
+package constraint
+
+import (
+	"context"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+const (
+	totalConstraintsName = "total_constraints"
+)
+
+var (
+	constraintsTotalM = stats.Int64(totalConstraintsName, "Total number of constraints", stats.UnitDimensionless)
+
+	enforcementActionKey = tag.MustNewKey("enforcement_action")
+	statusKey            = tag.MustNewKey("status")
+)
+
+func init() {
+	register()
+}
+
+func register() {
+	views := []*view.View{
+		{
+			Name:        totalConstraintsName,
+			Measure:     constraintsTotalM,
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{enforcementActionKey, statusKey},
+		},
+	}
+
+	if err := view.Register(views...); err != nil {
+		panic(err)
```

let's return this error to init and panic inside of init().